JS.md

```diff
@@ -0,0 +1,237 @@
+`f` in `m.exports`:
+* Let `index` be the function index of `f`.
+* If `f` is the first export of `index`, let `v` be a new 
+  [Exported Function](#exported-function-exotic-objects) Exotic Object given `instance` and `index`.
+* Otherwise, let `v` be the previously-created Exported Function Exotic Object.
+* Add a plain data property to `exportObject` whose property name is given by `f` and
+  whose value is `v`.
+
+Return a new `Wasm.Instance` instance containing `instance` where the initial value of
+the [`exports`](#wasminstance-exports-property) data property is
+`exportObject`.
+
+## Exported Function Exotic Objects
+
+Functions exported by WebAssembly modules are reflected in JS via a new kind
+of *Exported Function* 
+[Exotic Object](http://tc39.github.io/ecma262/#sec-built-in-exotic-object-internal-methods-and-slots).
+Like [Bound Function](http://tc39.github.io/ecma262/#sec-bound-function-exotic-objects) Exotic Object,
+Exported Functions do not have the normal function internal slots but instead have:
+ * [[WasmInstance]] : the [`Eval.instance`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/eval.ml#L15)
+   containing the exported function
+ * [[WasmFunctionIndex]] : the index of the function inside the module
+
+Exported Functions also have the following data properties:
+* the `length` property is set to the exported function's signature's arity 
+* the `name` is set to `index` as a Number value
+
+WebAssembly Exported Functions have a `[[Call]](this, argValues)` method defined as:
+ * Let `argTypes` be the list of value types defined by the signature of [[WasmFunctionIndex]].
+ * Let `args` be an empty list of coerced values.
+ * For each value type `t` in `argTypes` and value `v` in `argValues`:
+   * Append to `args` `v` coerced to `t` as follows:
+     * coerce `v` to `i32` via [`ToInt32(v)`](http://tc39.github.io/ecma262/#sec-toint32)
+     * throw a `TypeError` if `t` is `i64`
+     * coerce `v` to `f32` by first applying [`ToNumber(v)`](http://tc39.github.io/ecma262/#sec-tonumber)
+       and then converting the resulting IEEE754 64-bit double to a 32-bit float using `roundTiesToEven`
+     * coerce `v` to `f64` via [`ToNumber(v)`](http://tc39.github.io/ecma262/#sec-tonumber)
+ * Perform [`Eval.invoke`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/eval.ml#L327)
+   passing [[WasmInstance]], [[WasmFunctionIndex]], and `args`.
+ * Coerce the result of `Eval.invoke` as follows:
+   * if the return value is `None`, return `undefined`
+   * interpret `i32` as a signed integer and convert that integer to a Number value
+   * throw a `TypeError` if returning an `i64`
+   * return `f32`/`f64` as Number values, possibly performing
+     [canonicalization of NaNs](http://tc39.github.io/ecma262/#sec-setvalueinbuffer)
+
+Exported Functions do not have a [[Construct]] method and thus it is not possible to 
+call one with the `new` operator.
```

Comment:
To be pedantic, you might want to make Wasm functions a little less exotic by giving them a [[Construct]] method which is the same as their [[Call]] method, and throwing when new.target is not undefined. Otherwise, they will have subtly different behavior from ordinary JavaScript functions in a way which is an extra burden to implement but provides no real value.